File: program/program/find-sum-of-digits-of-a-number/find_sum_of_digits_of_a_number.py
# This python script calculates the sum of the digits of a random number n


def find_sum_of_digits_of_a_number():
    sum_of_digits = 0
    digits = input("Number: ")
    for digit in digits:
        sum_of_digits += int(digit)
    return sum_of_digits

print(f'Result: {find_sum_of_digits_of_a_number()}')


























# def sumOfDigits(n: int) -> int:
#     return sum([int(x) for x in str(n)])

File: program/program/find-sum-of-digits-of-a-number/find-sum-of-digits-of-a-number.jl
# Function to calculate the sum of digits of a number
function sum_of_digits(n::Int)
    sum = 0
    while n > 0
        sum += n % 10  # Add the last digit to sum
        n ÷= 10        # Remove the last digit from n
    end
    return sum
end

# Main code to get user input and call the function
println("Enter a number:")
n = parse(Int, readline())  # Reads input from the user and converts it to an integer
println("The sum of digits of ", n, " is ", sum_of_digits(n))
